drivers/sensor/nxp/mcux_lpcmp/mcux_lpcmp.c

```diff
@@ -273,8 +273,6 @@ static int mcux_lpcmp_sample_fetch(const struct device *dev, enum sensor_channel
 	const struct mcux_lpcmp_config *config = dev->config;
 	struct mcux_lpcmp_data *data = dev->data;
 
-	__ASSERT_NO_MSG(val != NULL);
```

Comment:
This fix looks generic- it should be the first commit in this PR. I think this was also already fixed upstream and can be removed, right?